# Détection de la position d'un usager

## Avec les positions des points d'accès

### Visualisation des fichiers utilisés

`Fichier contenant les positions des points d'accès ainsi que les forces de leurs signaux reçus par l'usager`

* X : position en (x) du point d'accès 
* Y : position en (y) du point d'accès 

In [1]:
import pandas as pd
df = pd.read_csv("./datas/with_acces/BSSID.csv", sep=";")
df

SSID              BSSID  Signal(%)     X     Y
0  eduroam  f4:ea:b5:b0:d4:25         90  3.04  6.37
1  eduroam  f4:ea:b5:dc:15:25         65  4.27  6.43
2  eduroam  f4:ea:b5:dc:6d:96        100  5.65  6.50
3  eduroam  f4:ea:b5:dc:15:25         85  5.77  8.03
4     THGA  f4:ea:b5:dc:2f:15         70  7.38  2.16

### Calcul de la position de l'usager

`Calcul de la position de l'usager depuis le fichier`

* Chargement du fichier des positions de points d'accès
* Calcul de la position proprement dite

In [2]:
from src.detect_user import detect_with_acces
from src.load import load_acces_points_2d

In [3]:
A, W = load_acces_points_2d("./datas/with_acces/BSSID.csv")
M = detect_with_acces(A, W)

In [4]:
print("[position détectée]  ")
print(f" M (x : {M[0]}, y : {M[1]})")

[position détectée]  
 M (x : 5.581107110963173, y : 5.202204830724139)


## Avec l'ensemble de positions des signaux collectés en avance

### Visualisation des fichiers utilisés

`Fichier contenant les signaux reçus par l'usager`

In [5]:
import pandas as pd
df = pd.read_csv("./datas/without_acces/BSSID_user.csv", sep=";")
df

SSID              BSSID  Signal(%)
0  eduroam  f4:ea:b5:b0:d4:25         90
1  eduroam  f4:ea:b5:dc:15:25         65
2  eduroam  f4:ea:b5:dc:6d:96         90
3     THGA  f4:ea:b5:dc:2f:15         70

`Fichier contenant les signaux collectés et les positions où ils ont été prises`

* X : position en (x) où a été collecté le signal
* Y : position en (y) où a été collecté le signal

In [6]:
df = pd.read_csv("./datas/without_acces/BSSIDs_collected.csv", sep=";")
df

SSID              BSSID  Signal(%)     X     Y
0                         eduroam  f4:ea:b5:b0:d4:25         90  3.04  6.37
1                         eduroam  f4:ea:b5:dc:6d:a5         15  3.04  6.37
2                         eduroam  f4:ea:b5:dc:15:25         70  3.04  6.37
3                         eduroam  f4:ea:b5:b0:d4:16        100  3.04  6.37
4                         eduroam  f4:ea:b5:dc:15:16         80  3.04  6.37
..                            ...                ...        ...   ...   ...
449  DIRECT-94-HP ENVY Photo 7800  16:cb:19:29:6f:94         20  2.95  1.78
450  DIRECT-94-HP ENVY Photo 7801  16:cb:19:29:6f:94         85  2.95  1.78
451             Vodafone Homespot  a2:e7:ae:36:cf:a0         14  2.95  1.78
452           ClickShare-Labor008  f4:6a:dd:30:1e:25         65  2.95  1.78
453                FRITZ!Box 3490  34:31:c4:c5:36:90         14  2.95  1.78

[454 rows x 5 columns]

On a `454` signaux récoltés dans le fichier csv

### Calcul de la position de l'usager

`Calcul de la position de l'usager depuis le fichier`

* Chargement du fichier des signaux de l'usager
* Chargement des signaux collectés avec les positions où ils ont été prises
* Calcul de la position proprement dite

In [7]:
from src.detect_user import detect_collected_sim
from src.load import load_collected_user, load_collected_positions_2d

In [8]:
E = load_collected_user("./datas/without_acces/BSSID_user.csv")
P, D = load_collected_positions_2d("./datas/without_acces/BSSIDs_collected.csv")
M = detect_collected_sim(E, P, D)

In [9]:
if M is not None:
    print("[Position détectée]  ")
    print(f" M (x : {M[0]}, y : {M[1]})")
else:
    print("[Aucune position détectée]  ")

[Position détectée]  
 M (x : 5.010830987562494, y : 7.287393508327559)
